subprojects/build-option/src/main/java/org/gradle/internal/buildoption/Origin.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.buildoption;
+
+import org.apache.commons.lang.StringUtils;
+import org.gradle.cli.CommandLineArgumentException;
+
+public abstract class Origin {
+    protected String source;
+
+    public static Origin withGradleProperty(String gradleProperty) {
```

Comment:
:o: I'd probably rename this to `forGradleProperty`. I reads more like "the origin of this source is a Gradle property".